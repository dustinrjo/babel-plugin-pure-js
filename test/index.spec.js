const babel = require('babel-core');
const plugin = require('../src');

describe('variable declaration', () => {
	it('prohibits var', () => {
		expect(() => babel.transform('var foo = 1;', { plugins: [plugin] }) ).toThrow();
	});
	it('prohibits let', () => {
		expect(() => babel.transform('let foo = 1;', { plugins: [plugin] }) ).toThrow();
	});
	it('allows const', () => {
		const { code } = babel.transform('const foo = 1;', { plugins: [plugin] });
		expect(code).toMatchSnapshot();
	});
});

describe('assignment', () => {
	it('prohibits variable assignment', () => {
		expect(() => babel.transform('foo = 1;', { plugins: [plugin] }) ).toThrow();
	});
	it('prohibits variable += assignment', () => {
		expect(() => babel.transform('foo += 1;', { plugins: [plugin] }) ).toThrow();
	});
	it('prohibits variable -= assignment', () => {
		expect(() => babel.transform('foo += 1;', { plugins: [plugin] }) ).toThrow();
	});
	it('prohibits property assignment', () => {
		expect(() => babel.transform('foo.bar = 1;', { plugins: [plugin] }) ).toThrow();
	});
	it('prohibits indexed assignment', () => {
		expect(() => babel.transform('foo["bar"] = 1;', { plugins: [plugin] }) ).toThrow();
	});
	it('prohibits array assignment', () => {
		expect(() => babel.transform('foo[0] = 1;', { plugins: [plugin] }) ).toThrow();
	});

	it('prohibits variable increment and decrement', () => {
		expect(() => babel.transform('foo++;', { plugins: [plugin] }) ).toThrow();
		expect(() => babel.transform('foo++;', { plugins: [plugin] }) ).toThrow();
		expect(() => babel.transform('++foo;', { plugins: [plugin] }) ).toThrow();
		expect(() => babel.transform('--foo;', { plugins: [plugin] }) ).toThrow();
	});
});

describe('this', () => {
	it('prohibits this', () => {
		expect(() => babel.transform('this.foo();', { plugins: [plugin] }) ).toThrow();
	});
});
