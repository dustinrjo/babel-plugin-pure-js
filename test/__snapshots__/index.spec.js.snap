// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`variable declaration allows const 1`] = `"const bar = 1;"`;
